cardano-api/src/Cardano/Api/SerialiseRaw.hs

```diff
@@ -31,9 +32,28 @@ serialiseToRawBytesHex = Base16.encode . serialiseToRawBytes
 serialiseToRawBytesHexText :: SerialiseAsRawBytes a => a -> Text
 serialiseToRawBytesHexText = Text.decodeUtf8 . serialiseToRawBytesHex
 
-deserialiseFromRawBytesHex :: SerialiseAsRawBytes a
-                           => AsType a -> ByteString -> Maybe a
-deserialiseFromRawBytesHex proxy hex =
-    case Base16.decode hex of
-      Right raw -> deserialiseFromRawBytes proxy raw
-      Left _msg -> Nothing
+data RawBytesHexError
+  = RawBytesHexErrorBase16DecodeFail
+      ByteString -- ^ original input
+      String -- ^ error message
```

Comment:
Can we not do better than a String here?

Reply:
Sorry? Too much indirection for my English.

Do you suggest having something more structured than a String in the second field of RawBytesHexErrorBase16DecodeFail or not?